src/sdk/trace/span_exporter.zig

```diff
@@ -0,0 +1,30 @@
+const pb = @import("../../opentelemetry/proto/trace/v1.pb.zig");
+
+/// SpanExporter is the interface that provides an
+pub const SpanExporter = struct {
+    // std.mem.Allocator's style
+    ptr: *anyopaque,
+    vtable: *const VTable,
+
+    const Self = @This();
+
+    /// VTable defines the methods that the SpanExporter's instance must implement.
+    pub const VTable = struct {
+        /// exportSpans is the method that export a batch of spans.
+        /// NOTE: In other languages, the span types is ReadOnlySpan for improving stability.
+        /// but it is not defined in the OpenTelemetry specification, so for now we don't use it.
+        exportSpansFn: *const fn (
+            ctx: *anyopaque,
+            spans: []pb.Span,
```

Comment:
I think I made a mistake in the metrics, see my reasoning in #11.
I am not sure coupling to the protobuf data structures here is a goo idea either.

Ideally, we would have an internal representation of Spans, and then only convert to protobuf in the `otlp` exporter, which is self-contained.
Making protobuf tyoes part of the interface method here will force us to couple the entire SDK to them, which is not desirable.

WDYT?

Reply:
I strongly agree.
I will define another representation of Span.